src/layers/tile/finite.rs

```diff
@@ -51,9 +51,9 @@ impl FiniteTileLayerData {
         })
     }
 
-    pub(crate) fn get_tile(&self, x: usize, y: usize) -> Option<&LayerTileData> {
-        if x < self.width as usize && y < self.height as usize {
-            self.tiles[x + y * self.width as usize].as_ref()
+    pub(crate) fn get_tile(&self, x: i32, y: i32) -> Option<&LayerTileData> {
+        if x < self.width as i32 && y < self.height as i32 && x >= 0 && y >= 0 {
+            self.tiles[(x + y * self.width as i32) as usize].as_ref()
```

Comment:
I think it would be better to convert to `usize` before multiplying, since the i32 has a narrower range. So this should probably be something like:
```suggestion
            self.tiles[x as usize + y as usize * self.width].as_ref()
```